File: Examples/CSharp/ModifyingAndConvertingImages/BinarizationWithFixedThreshold.cs
// GIST-ID: 5ad52a4d7d6d5df55cb1bcc7e17deaa9
/*
This project uses the Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API references 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

using System;
namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class BinarizationWithFixedThreshold
    {
        public static void Run()
        {
            Console.WriteLine("Running example BinarizationWithFixedThreshold");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Load an image into an instance of Image
            using (Image image = Image.Load(dataDir + "aspose-logo.jpg"))
            {
                // Cast the image to RasterCachedImage and check if the image is cached                
                RasterCachedImage rasterCachedImage = (RasterCachedImage)image;
                if (!rasterCachedImage.IsCached)
                {
                    // Cache the image if it is not already cached
                    rasterCachedImage.CacheData();
                }

                // Binarize the image with a predefined fixed threshold and save the resultant image                
                rasterCachedImage.BinarizeFixed(100);
                rasterCachedImage.Save(dataDir + "BinarizationWithFixedThreshold_out.jpg");
            }

            Console.WriteLine("Finished example BinarizationWithFixedThreshold");
        }
    }
}

File: Examples/VisualBasic/ModifyingAndConvertingImages/BinarizationWithFixedThreshold.vb
Imports Aspose.Imaging.ImageFilters.FilterOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Class BinarizationWithFixedThreshold
        Public Shared Sub Run()
            ' ExStart:BinarizationWithFixedThreshold
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Load an image in an instance of Image
            Using image__1 As Image = Image.Load(dataDir & Convert.ToString("aspose-logo.jpg"))
                ' Cast the image to RasterCachedImage and  Check if image is cached
                Dim rasterCachedImage As RasterCachedImage = DirectCast(image__1, RasterCachedImage)
                If Not rasterCachedImage.IsCached Then
                    ' Cache image if not already cached
                    rasterCachedImage.CacheData()
                End If
                ' Binarize image with predefined fixed threshold and Save the resultant image
                rasterCachedImage.BinarizeFixed(100)
                rasterCachedImage.Save(dataDir & Convert.ToString("BinarizationWithFixedThreshold_out.jpg"))
            End Using
            ' ExEnd:BinarizationWithFixedThreshold
        End Sub
    End Class
End Namespace